test/jdk/javax/swing/JTable/PrintAllPagesTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * @test
+ * @bug 8257810
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary  Verifies if all pages are printed if scrollRectToVisible is set.
+ * @run main/manual PrintAllPagesTest
+ */
+import java.awt.print.PrinterJob;
+import java.awt.print.PrinterException;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.AbstractTableModel;
+import javax.swing.table.TableModel;
+import javax.swing.WindowConstants;
+
+public class PrintAllPagesTest {
+    static JFrame f;
+    static JTable table;
+    static volatile boolean ret = false;
+
+    static final String INSTRUCTIONS = """
+         Note: You must have a printer installed for this test.
+         If printer is not available, the test passes automatically.
+
+         A JTable with 1000 rows and a print dialog will be shown.
+         If only 1 page is printed,
+         then press fail else press pass.
+            """;
+
+    public static void main(String[] args) throws Exception {
+
+        PrinterJob pj = PrinterJob.getPrinterJob();
+        if (pj.getPrintService() == null) {
+            System.out.println("Printer not configured or available."
+                    + " Test cannot continue.");
+            PassFailJFrame.forcePass();
+        }
+
+        PassFailJFrame passFailJFrame = new PassFailJFrame(INSTRUCTIONS);
+
+        SwingUtilities.invokeAndWait(() -> {
+            printAllPagesTest();
+        });
+
+        // add the test frame to dispose
+        PassFailJFrame.addTestFrame(f);
+
+        // Arrange the test instruction frame and test frame side by side
+        PassFailJFrame.positionTestFrame(f, PassFailJFrame.Position.HORIZONTAL);
+
+        SwingUtilities.invokeAndWait(() -> {
+            try {
+                ret = table.print();
+            } catch (PrinterException ex) {
+                ret = false;
+            }
+        });
+
+        if (!ret) {
+            throw new RuntimeException("Printing cancelled/failed");
+        }
+        passFailJFrame.awaitAndCheck();
+    }
+
+
+    private static void printAllPagesTest() {
+        TableModel dataModel = new AbstractTableModel() {
+            @Override
+            public int getColumnCount() {
+                return 10;
+            }
+
+            @Override
+            public int getRowCount() {
+                return 1000;
+            }
+
+            @Override
+            public Object getValueAt(int row, int col) {
+                return Integer.valueOf(0 == col ? row + 1 : row * col);
+            }
+        };
+        table = new JTable(dataModel);
+        JScrollPane scrollpane = new JScrollPane(table);
+        table.scrollRectToVisible(table.getCellRect(table.getRowCount() - 1, 0, false));
+
+        f = new JFrame("Table test");
+        f.add(scrollpane);
+        f.setSize(1000, 800);
+        f.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
```

Comment:
setDefaultCloseOperation() is required.....?

Reply:
Good catch! Not really… It must not be `EXIT_ON_CLOSE`. The default value is `HIDE_ON_CLOSE` and it would work well because the frame will be disposed of after the latch is released.

And here I see another problem: the frame that's created by test has to handle `WindowListener.windowClosing`. Since it's required by all tests, `PassFailJFrame` should provide a default listener for this purpose.

Reply:
There are many swing tests that has EXIT_ON_CLOSE so not sure if it's a problem .There are couple which have DISPOSE_ON_CLOSE which I can changed to but not sure if it will have any effect..

Reply:
It is a problem for jtreg tests: `System.exit` shouldn't be called. We should not use it for new tests even if there are tests which use it.

`DISPOSE_ON_CLOSE` is fine: it disposes of the frame; once there are no frames left, the JVM exits.

For this test, `DO_NOTHING_ON_CLOSE` could also work well. After all, the user is not expected to interact with the secondary frame at all.

Reply:
I dont think it is a problem now..As the tests are now run in othervm mode and not in samevm mode as we used to do pre2018..
there's a comment on this aspect too https://github.com/openjdk/jdk/pull/8293#discussion_r854488853

ANyway, I have removed the contentious call